File: mobile/types/book.ts
export type BookLanguage = "english" | "malayalam" | "hindi";
export type BookCategory = "health" | "wealth" | "happiness" | "mindfulness";
export type BookAccessType = "free" | "premium";
export type BookStatus = "draft" | "upcoming" | "live";

export type Book = {
  _id: string;
  title: string;
  description: string;
  language: BookLanguage;
  category: BookCategory;
  coverImageUrl: string;
  coverPublicId?: string;
  introAudioUrl: string;
  introAudioPublicId?: string;
  introAudioFileName?: string;
  totalDurationSeconds: number;
  accessType: BookAccessType;
  status: BookStatus;
  isFeatured: boolean;
  isTrending: boolean;
  sortOrder: number;
  tags: string[];
};

export type Chapter = {
  _id: string;
  bookId: string;
  title: string;
  chapterNumber: number;
  description: string;
  audioUrl: string | null;
  durationSeconds: number;
  isFreePreview: boolean;
  locked?: boolean;
};

File: flutter_app/lib/features/books/domain/entities/book.dart
class Book {
  const Book({
    required this.id,
    required this.title,
    required this.description,
    required this.language,
    required this.category,
    required this.coverImageUrl,
    required this.audioUrl,
    required this.accessType,
    required this.status,
    required this.duration,
    this.episodes = const [],
  });
  final String id,
      title,
      description,
      language,
      category,
      coverImageUrl,
      audioUrl,
      accessType,
      status;
  final int duration;
  final List<BookEpisode> episodes;
}

class BookEpisode {
  const BookEpisode({required this.title, required this.audioUrl, this.description = '', this.order = 0});
  final String title, audioUrl, description;
  final int order;
}
